pf_driver/src/pf/pf_interface.cpp

```diff
@@ -328,3 +373,19 @@ void PFInterface::reconfig_callback_r2300(pf_driver::PFDriverR2300Config& config
   params_ = protocol_interface_->get_scan_parameters(config_.start_angle);
   pipeline_->set_scan_params(params_);
 }
+
+bool const PFInterface::is_scanner_accessible()
+{
+  std::string command = "ping -c 1 -W 1 " + ip_ + " > /dev/null 2>&1";
```

Comment:
:scream: 

I guess this is the horrible hack that you were talking about...

Reply:
Please add a comment that explains the ping arguments.

Reply:
do you have to use ping, or could you just use the `protocol_interface_->get_protocol_info();` call that is also used in `init()`?